connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -419,7 +423,14 @@ private <T> Collection<PluginDesc<T>> getServiceLoaderPluginDesc(Class<T> klass,
         Collection<PluginDesc<T>> result = new ArrayList<>();
         try {
             ServiceLoader<T> serviceLoader = ServiceLoader.load(klass, loader);
-            for (T pluginImpl : serviceLoader) {
+            for (Iterator<T> iterator = serviceLoader.iterator(); iterator.hasNext(); ) {
+                T pluginImpl;
+                try {
+                    pluginImpl = iterator.next();
+                } catch (ServiceConfigurationError t) {
+                    log.error("Unable to instantiate plugin{}", reflectiveErrorDescription(t.getCause()), t);
```

Comment:
Would it be more informative to use the type of plugin class being instantiated instead of just "plugin"?
```suggestion
                    log.error("Unable to instantiate {}{}", klass.getSimpleName(), reflectiveErrorDescription(t.getCause()), t);
```

Reply:
I changed this and the previous log message so that it wasn't confusing that we were instantiating a SinkConnector.
It is unfortunate that on this branch we don't get to see the `Class<?>` after a failure occurs, and have to rely on the ServiceConfigurationError message to report what specific plugin had an issue.

Reply:
Works for me 👍